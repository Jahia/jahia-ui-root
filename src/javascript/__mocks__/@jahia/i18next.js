var reacti18next = require('react-i18next');

module.exports = {
    getI18n: reacti18next.getI18n
};
