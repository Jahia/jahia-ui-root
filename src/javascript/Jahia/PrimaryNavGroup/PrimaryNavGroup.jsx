import React, {Suspense} from 'react';
import PropTypes from 'prop-types';
import {PrimaryNavItem, PrimaryNavItemsGroup} from '@jahia/moonstone';
import {registry} from '@jahia/ui-extender';
import {useTranslation} from 'react-i18next';
import {useHistory} from 'react-router';
import {useNodeInfo} from '@jahia/data-helper';
import {useSelector} from 'react-redux';

export const PrimaryNavGroup = ({isDisplayedWhenCollapsed, target}) => {
    const {t} = useTranslation();
    const history = useHistory();
    const current = useSelector(state => ({language: state.language, site: state.site}));
    const navItems = registry.find({type: 'primary-nav-item', target});
    const requiredPermission = ['jcr:read_default'];
    const requiredPaths = [];
    navItems.filter(navItem => navItem.requiredPermission !== undefined).forEach(navItem => {
        if (!requiredPermission.includes(navItem.requiredPermission)) {
            requiredPermission.push(navItem.requiredPermission);
        }

        if (navItem.requiredPermissionPath !== undefined && !requiredPaths.includes(navItem.requiredPermissionPath)) {
            requiredPaths.push(navItem.requiredPermissionPath);
        }
    });
    if (requiredPaths.length === 0) {
        requiredPaths.push('/sites/' + current.site);
    }

    const permissions = useNodeInfo({paths: requiredPaths, language: 'en'},
        {getPermissions: requiredPermission});
    if (permissions.loading === true || permissions.nodes === null) {
        return null;
    }

    console.log(permissions);

    return (
        <Suspense fallback="loading...">
            <PrimaryNavItemsGroup isDisplayedWhenCollapsed={isDisplayedWhenCollapsed}>
                {navItems.filter(navItem => navItem.requiredPermission === undefined ||
                    permissions.nodes.find(node => {
                        if (navItem.requiredPermissionPath !== undefined) {
                            return node.path === navItem.requiredPermissionPath;
                        }

                        return node.path === '/sites/' + current.site;
                    })[navItem.requiredPermission]).map(item => {
                    if (item.render) {
                        return React.cloneElement(item.render(), {key: item.key});
                    }

                    return (
                        <PrimaryNavItem key={item.key}
                                        role={`${item.key}-menu-item`}
                                        isSelected={history.location.pathname.startsWith(item.path)}
                                        icon={item.icon}
                                        label={t(item.label)}
                                        onClick={() => history.push(item.path)}/>
                    );
                })}
            </PrimaryNavItemsGroup>
        </Suspense>
    );
};

PrimaryNavGroup.defaultProps = {
    isDisplayedWhenCollapsed: true
};

PrimaryNavGroup.propTypes = {
    isDisplayedWhenCollapsed: PropTypes.bool,
    target: PropTypes.string.isRequired
};

export default PrimaryNavGroup;
