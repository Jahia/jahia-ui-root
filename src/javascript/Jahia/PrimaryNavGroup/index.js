import PrimaryNavGroup from './PrimaryNavGroup';

export default PrimaryNavGroup;
