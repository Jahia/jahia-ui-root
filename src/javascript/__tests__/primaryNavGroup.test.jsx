import React from 'react';
import renderer from 'react-test-renderer';
import PrimaryNavGroup from '../Jahia/PrimaryNavGroup';
import {registry} from '@jahia/ui-extender';

describe('primary nav group', () => {

    test('collapsed and empty nav bar', () => {

        const component = renderer.create(
            <PrimaryNavGroup target={'none'} isDisplayedWhenCollapsed={true}/>,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('expanded and empty nav bar', () => {

        const component = renderer.create(
            <PrimaryNavGroup target={'none'} isDisplayedWhenCollapsed={false}/>,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('expanded and not empty nav bar', () => {
        registry.add('primary-nav-item', 'test1', {
            targets:['target1'],
            render: () => <span>mock_content</span>
        });

        const component = renderer.create(
            <PrimaryNavGroup target="target1" isDisplayedWhenCollapsed={false}/>,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('collapsed and not empty nav bar', () => {
        registry.add('primary-nav-item', 'test2', {
            targets:['target2'],
            render: () => <span>mock_content</span>
        });

        const component = renderer.create(
            <PrimaryNavGroup target="target2" isDisplayedWhenCollapsed={true}/>,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
