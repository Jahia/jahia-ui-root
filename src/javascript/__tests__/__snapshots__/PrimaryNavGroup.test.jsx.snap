// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primary nav group collapsed and empty nav bar 1`] = `
Array [
  <li
    className="Zbf5IjDJoFOYgLMhuYJX3"
  >
    <hr
      className="AzU6OD_KpO59W4OjMaKrt _1nz23RxJRG8lzjw4UEkam1 _3ehD-Vp1xzL3c4pv4NXQRU _1c9yGr-7IZ-IwQq0y_q_IK"
    />
  </li>,
  <li
    className="Zbf5IjDJoFOYgLMhuYJX3"
  >
    <ul />
  </li>,
]
`;

exports[`primary nav group collapsed and not empty nav bar 1`] = `
Array [
  <li
    className="Zbf5IjDJoFOYgLMhuYJX3"
  >
    <hr
      className="AzU6OD_KpO59W4OjMaKrt _1nz23RxJRG8lzjw4UEkam1 _3ehD-Vp1xzL3c4pv4NXQRU _1c9yGr-7IZ-IwQq0y_q_IK"
    />
  </li>,
  <li
    className="Zbf5IjDJoFOYgLMhuYJX3"
  >
    <ul>
      <span>
        mock_content
      </span>
    </ul>
  </li>,
]
`;

exports[`primary nav group expanded and empty nav bar 1`] = `null`;

exports[`primary nav group expanded and not empty nav bar 1`] = `null`;
